boards/arm/96b_carbon/96b_carbon.dts

```diff
@@ -16,6 +16,37 @@
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
 	};
+
+	leds {
+		compatible = "gpio-leds";
```

Comment:
I'm still learning about how Zephyr uses DT, but I can't find a property in dts/bindings that has a compatible constraint "gpio-leds". What are the rules here; does this need to be written too or can compatible properties be set in dts files even if they don't have accompanying YAML bindings in Zephyr?

I've looked at extract_dts_includes.py and I've read http://docs.zephyrproject.org/devices/dts/device_tree.html, but I'm still confused.

Reply:
@mbolivar: You're right, a yaml binding file is required to extract information.
Besides, some script changes are also required to get #defines generation working. I'm working on it right now.
Very next step would be to be able to generate full code as proposed for pinmux (#5021). This whole task might take some while, then I'd like to be able to split it in two and having this first dts part merged while I complete generation part. Does its sound ok to you?

Reply:
@mbolivar:Thanks for the feedback. FYI, in the end I updated scope of current RFC to cover definitions generation as well.